src/storage-types/src/sources/mysql.rs

```diff
@@ -159,3 +167,139 @@ impl RustType<ProtoMySqlSourceDetails> for MySqlSourceDetails {
         })
     }
 }
+
+/// Represents a MySQL transaction id
+#[derive(Debug, Clone, Copy, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum TransactionId {
```

Comment:
`TransactionId::Absent` feels like the transaction id itself is absent. It'd be great to find a short descriptive name for this type. `ServerState` is slightly smaller and I think better matches the semantics but maybe there is an even better one. `Source` is unfortunately a severely overloaded term in materialize :sweat_smile:

Reply:
How about `GTIDState`? `GTIDState::Absent` means we haven't seen any GTIDs with this source-id, and `GTIDState::Next` represents the next transaction-id we expect to see for this source.